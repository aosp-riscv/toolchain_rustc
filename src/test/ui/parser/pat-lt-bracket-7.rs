fn main() {
    struct Thing(u8, [u8; 0]);
    let foo = core::iter::empty();

<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
    for Thing(x[]) in foo {} //~ ERROR: expected one of `)`, `,`, or `@`, found `[`
=======
    for Thing(x[]) in foo {}
    //~^ ERROR: expected one of `)`, `,`, `@`, or `|`, found `[`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
}

const RECOVERY_WITNESS: () = 0; //~ ERROR mismatched types
