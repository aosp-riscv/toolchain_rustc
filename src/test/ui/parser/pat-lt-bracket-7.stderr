<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
error: expected one of `)`, `,`, or `@`, found `[`
=======
error: expected one of `)`, `,`, `@`, or `|`, found `[`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
  --> $DIR/pat-lt-bracket-7.rs:5:16
   |
LL |     for Thing(x[]) in foo {}
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
   |                ^ expected one of `)`, `,`, or `@` here
=======
   |                ^ expected one of `)`, `,`, `@`, or `|` here
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)

error[E0308]: mismatched types
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
  --> $DIR/pat-lt-bracket-7.rs:8:30
=======
  --> $DIR/pat-lt-bracket-7.rs:9:30
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
   |
LL | const RECOVERY_WITNESS: () = 0;
   |                              ^ expected (), found integer
   |
   = note: expected type `()`
              found type `{integer}`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
